pkgs/games/freeorion/default.nix

```diff
@@ -0,0 +1,23 @@
+{ stdenv, fetchgit, cmake, boost, SDL2, python2, freetype, openal, libogg, libvorbis, zlib, libpng, libtiff, libjpeg, mesa, glew, git, doxygen }:
+
+stdenv.mkDerivation rec {
+  version = "2016-04-16";
+  name = "freeorion-${version}";
+
+  src = fetchgit {
+    url = "https://github.com/freeorion/freeorion";
```

Comment:
Oh, sorry, I understood it the wrong way.

I think that using release version with the patch is better, but if cherry-picking the patch is hard and/or it doesn't fix the problem using git is fine.

Unfortunately, leaving `.git` deterministically is very hard (see https://github.com/NixOS/nixpkgs/issues/8567), so basically when you use `leaveDotGit` SHA checksum is destined to change. So, unless it's _really_ needed, we try to avoid this.

For another possible solution see for example [ejabberd](https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/xmpp/ejabberd/default.nix#L18) -- you can provide a fake _git_ to the build script which would output something fixed. I've done it this way:

1. First, make this `git` output all arguments thrown at it and see what the build script actually calls;
2. Second, hard-code answers to specific combinations of arguments.

This may be easier than patching the build script (that is, if you decide to use git version).